File: app/src/main/java/com/fonteraro/martaco/cinemahub/ui/details/DetailsActivity.java
package com.fonteraro.martaco.cinemahub.ui.details;

import static com.fonteraro.martaco.cinemahub.utils.AppConstants.EXTRA_KEY_MOVIE_ID;

import android.os.Bundle;
import com.fonteraro.martaco.cinemahub.BR;
import com.fonteraro.martaco.cinemahub.R;
import com.fonteraro.martaco.cinemahub.databinding.ActivityDetailsBinding;
import com.fonteraro.martaco.cinemahub.ui.base.BaseActivity;
import javax.inject.Inject;
import androidx.annotation.Nullable;

public class DetailsActivity extends BaseActivity<ActivityDetailsBinding, DetailsViewModel> {

    @Inject
    DetailsViewModel detailsViewModel;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        getViewDataBinding().setLifecycleOwner(this);
        fetchMovie();
    }

    private void fetchMovie() {
        long movieId = retrieveMovieId();
        if (movieId != -1) getViewModel().getMovieDetails(movieId);
    }

    private Long retrieveMovieId() {
        return getIntent().getLongExtra(EXTRA_KEY_MOVIE_ID, -1);
    }

    @Override
    public int getBindingVariable() {
        return BR.viewModel;
    }

    @Override
    public int getLayoutId() {
        return R.layout.activity_details;
    }

    @Override
    public DetailsViewModel getViewModel() {
        return detailsViewModel;
    }
}

File: app/src/main/java/com/fonteraro/martaco/cinemahub/ui/details/DetailsActivity.kt
package com.fonteraro.martaco.cinemahub.ui.details

import com.fonteraro.martaco.cinemahub.ui.base.BaseActivity
import androidx.lifecycle.ViewModelProvider
import android.os.Bundle
import androidx.activity.viewModels
import com.fonteraro.martaco.cinemahub.BR
import com.fonteraro.martaco.cinemahub.utils.AppConstants
import com.fonteraro.martaco.cinemahub.R
import com.fonteraro.martaco.cinemahub.databinding.ActivityDetailsBinding
import javax.inject.Inject

class DetailsActivity : BaseActivity<ActivityDetailsBinding, DetailsViewModel>() {
    @Inject
    lateinit var viewModelFactory: ViewModelProvider.Factory

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        viewDataBinding!!.lifecycleOwner = this
        fetchMovie()
    }

    private fun fetchMovie() {
        val movieId = retrieveMovieId()
        if (movieId != -1L) viewModel.getMovieDetails(movieId)
    }

    private fun retrieveMovieId(): Long {
        return intent.getLongExtra(AppConstants.EXTRA_KEY_MOVIE_ID, -1)
    }

    override fun getBindingVariable(): Int {
        return BR.viewModel
    }

    override fun getLayoutId(): Int {
        return R.layout.activity_details
    }

    override fun getViewModel(): DetailsViewModel {
        val detailsViewModel: DetailsViewModel by viewModels { viewModelFactory }
        return detailsViewModel
    }
}
